internal/namespace/controllers/reconciler.go

```diff
@@ -0,0 +1,87 @@
+package namespace
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	hub "github.com/kubeslice/worker-operator/internal/hub/hubclient"
+	"github.com/kubeslice/worker-operator/internal/logger"
+	"github.com/kubeslice/worker-operator/pkg/events"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+//+kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list;watch;create;update;patch;delete
+
+type Reconciler struct {
+	client.Client
+	EventRecorder *events.EventRecorder
+	Scheme        *runtime.Scheme
+	Log           logr.Logger
+	Hubclient     *hub.HubClientConfig
+}
+
+var excludedNs = []string{"kube-system", "default", "kubeslice-system", "kube-node-lease",
+	"kube-public", "istio-system"}
+
+func (c *Reconciler) getSliceNameFromNs(ns string) (string, error) {
+	namespace := corev1.Namespace{}
+	err := c.Client.Get(context.Background(), types.NamespacedName{Name: ns}, &namespace)
+	if err != nil {
+		c.Log.Error(err, "error while retrieving namespace")
+		return "", err
+	}
+	// TODO: label should not be hardcoded, it should come as a constant from slice reconciler pkg
+	return namespace.Labels["kubeslice.io/slice"], nil
```

Comment:
this will crash if the ns does not have any labels on it, dereferencing a null map.

Reply:
If the labels are present please check if the slice label is present. test for presence:

sliceName, present := labels[""]
if !present {
   return "", nil
}

return sliceName, nil

Reply:
No no it won't crash, it will just return an empty string. I have tested that scenario as well

Reply:
Please see this example - https://go.dev/play/p/W7-bqAImRLc
even if it is present or not we will be adding an empty string only, so we don't need to check whether the key is present or not in map

Reply:
I have added a new commit to handle the case when namespace labels are nil.